build_projects/shared-build-targets-utils/Publishing/AzurePublisher.cs

```diff
@@ -183,12 +183,37 @@ public string CalculateArchiveUploadUrl(string archiveFile, string channel, stri
             return $"{s_dotnetBlobRootUrl}{CalculateArchiveBlob(archiveFile, channel, version)}";
         }
 
-        public string CalculateArchiveBlob(string archiveFile, string channel, string version)
+        public static string CalculateArchiveBlob(string archiveFile, string channel, string version)
         {
             return $"{channel}/Binaries/{version}/{Path.GetFileName(archiveFile)}";
         }
 
-        public void DownloadFiles(string blobVirtualDirectory, string fileExtension, string downloadPath)
+        public static void DownloadFile(string blobFilePath, string localDownloadPath)
+        {
+            var blobUrl = $"{s_dotnetBlobRootUrl}{blobFilePath}";
+
+            using (var client = new HttpClient())
+            {
+                client.Timeout = TimeSpan.FromMinutes(1);
+
+                var request = new HttpRequestMessage(HttpMethod.Get, blobUrl);
+                var sendTask = client.SendAsync(request, HttpCompletionOption.ResponseHeadersRead);
+                var response = sendTask.Result.EnsureSuccessStatusCode();
+
+                var httpStreamTask = response.Content.ReadAsStreamAsync();
+                //httpStreamTask.Wait();
+
+                var httpStream = httpStreamTask.Result;
```

Comment:
In my general opinion, these "utils" should be async as needed.  And then force the callers to deal with the async behavior.

Someday I could envision a Build.Framework that supported async Targets, and then it would all flow nicely.  (Of course it is going away, so that won't happen).